src/ints/bios_keyboard.cpp

```diff
@@ -27,15 +27,6 @@
 #include "dos_inc.h"
 #include "SDL.h"
 
-/* SDL by default treats numlock and scrolllock different from all other keys.
- * In recent versions this can disabled by a environment variable which we set in sdlmain.cpp
- * Define the following if this is the case */
-#if SDL_VERSION_ATLEAST(1, 2, 14)
-#define CAN_USE_LOCK 1
-/* For lower versions of SDL we also use a slight hack to get the startup states of numclock and capslock right.
- * The proper way is in the mapper, but the repeating key is an unwanted side effect for lower versions of SDL */
-#endif
```

Comment:
> fullscreen 	vsync 	resolution 	output 	renderer 	Linux 	macOS 	Windows

Two interesting additions:
- Windows with `output=texture` supports `renderer=direct3d` - I tested it briefly on Windows 10 before sending PR - it works as well as `output=texture`, `renderer=opengl` and both renderers seem to support `vsync` correctly (no tearing!). I don't have access to the machine with Windows ATM though. I haven't tested if it solved the tearing on Linux yet, but vsync working correctly on one platform allows us to test-enable adaptive sync support via SDL2.
- I have no idea if it works, but NY00123 patch has allowed `renderer=metal` for macOS! (I kept it in)

Reply:
That's great news there are even more options available! 

I'm really impressed with how robust they've made SDL2, along with the modern X video drivers and latest kernel support for Intel and NVidia GPUs. I (had!) long since given up on full-screen modes for fear of getting trapped in a custom resolution, being punted back to the console, or out-right hangs. The dropped-keystroke issues with SDL 1.2 with Xorg were the last straw for DOSBox on my desktop relegating it my X-less Raspberry Pi 3. 

`dosbox-staging` has turned all of that around!  I've since tested a dizzying number of combinations on my x86_64 desktop: custom low-resolutions, various refresh rates, vsync on and off, and repeated flipping between full and desktop/windowed resolution. SDL2 has been bullet-proof.